amazon_kclpy/kcl.py

```diff
@@ -220,6 +220,17 @@ def shutdown(self, checkpointer, reason):
         '''
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def shutdownRequested(self, checkpointer):
+        '''
+                Called by a KCLProcess instance to indicate that this record processor is being shutdown.
+                And it gives an opportunity for record processor to checkpoint before shutdown.
+
+                :type checkpointer: amazon_kclpy.kcl.Checkpointer
+                :param checkpointer: A checkpointer which accepts a sequence number or no parameters.
+                '''
+        raise NotImplementedError
```

Comment:
This should probably do nothing.  If someone upgrades to the newer version without implementing the shutdownRequested it would cause their application to fail.